### PR TITLE
contracts: Use the log crate to output contract generated messages

### PR DESCRIPTION
**Pull request by athei**
This tiny PR has two benefits over the old approach which used `sp_runtime::print`:

- The messages are visible without changing the log level
- They are scoped properly under the `runtime::contracts` target

Also added a section to the README which explains the situation.

### CODE REVIEW

**Review by ascjones — APPROVED** [review]
:heart:

**Review by bkchr — APPROVED** [review]

File: frame/contracts/README.md [hunk 1]
```diff
@@ -57,4 +57,19 @@ will make things a lot easier. One such language is [`ink`](https://github.com/p
...
+`seal_println` API. This is exposed in ink! via
+[`ink_env::debug_println()`](https://docs.rs/ink_env/latest/ink_env/fn.debug_println.html).
+
+In order to see these messages the log level for the `runtime::contracts` target needs to be raised
+to at least the `info` level which is the default. However, those messages are easy to overlook
+because of the noise generated by block production. A good starting point for contract debugging
+could be:
+
+```bash
+cargo run --release -- --dev --tmp -lerror,runtime::contracts
```
Comment (bkchr) [inline frame/contracts/README.md:72]: Info logging is enabled by default for substrate. Maybe we should change it to log::debug?

Comment (athei) [reply frame/contracts/README.md:72]: Are you sure about this? I ran the node without any logging arguments and the message came out.

Comment (bkchr) [reply frame/contracts/README.md:72]: Ahh. Now I see what you are doing here. Should not have checked this on my phone :P 

But also in general this is more some debug out or not?

Comment (bkchr) [reply frame/contracts/README.md:72]: Okay, if they are reject on productive chains, this is fine

**Comment by athei** [comment]
bot merge

**Pull request merged**